src/main/java/se/michaelthelin/spotify/SpotifyHttpRequestRetryStrategy.java

```diff
@@ -0,0 +1,224 @@
+   * <li>ConnectionClosedException</li>
+   * <li>NoRouteToHostException</li>
+   * <li>SSLException</li>
+   * </ul>
+   * <p>
+   * and retriable HTTP status codes:<br>
+   * <ul>
+   * <li>SC_TOO_MANY_REQUESTS (429)</li>
+   * <li>SC_SERVICE_UNAVAILABLE (503)</li>
+   * </ul>
+   *
+   * @param maxRetries           how many times to retry; 0 means no retries
+   * @param defaultRetryInterval the default retry interval between
+   *                             subsequent retries if the {@code Retry-After} header is not set
+   *                             or invalid.
+   */
+  public SpotifyHttpRequestRetryStrategy(
+    final int maxRetries,
+    final TimeValue defaultRetryInterval) {
+    this(maxRetries, defaultRetryInterval,
+      Arrays.asList(
+        InterruptedIOException.class,
+        UnknownHostException.class,
+        ConnectException.class,
+        ConnectionClosedException.class,
+        NoRouteToHostException.class,
+        SSLException.class),
+      List.of(
+        HttpStatus.SC_SERVICE_UNAVAILABLE));
+  }
+
+  /**
+   * Create the HTTP request retry strategy with a max retry count of 1,
+   * default retry interval of 1 second, and using the following list of
+   * non-retriable I/O exception classes:<br>
+   * <ul>
+   * <li>InterruptedIOException</li>
+   * <li>UnknownHostException</li>
+   * <li>ConnectException</li>
+   * <li>ConnectionClosedException</li>
+   * <li>SSLException</li>
+   * </ul>
+   * <p>
+   * and retriable HTTP status codes:<br>
+   * <ul>
+   * <li>SC_TOO_MANY_REQUESTS (429)</li>
+   * <li>SC_SERVICE_UNAVAILABLE (503)</li>
+   * </ul>
+   */
+  public SpotifyHttpRequestRetryStrategy() {
+    this(1, TimeValue.ofSeconds(1L));
+  }
+
+  @Override
+  public boolean retryRequest(
+    final HttpRequest request,
+    final IOException exception,
+    final int execCount,
+    final HttpContext context) {
+    Args.notNull(request, "request");
+    Args.notNull(exception, "exception");
+
+    if (execCount > this.maxRetries) {
+      // Do not retry if over max retries
+      return false;
+    }
+    if (this.nonRetriableIOExceptionClasses.contains(exception.getClass())) {
+      return false;
+    } else {
+      for (final Class<? extends IOException> rejectException : this.nonRetriableIOExceptionClasses) {
+        if (rejectException.isInstance(exception)) {
+          return false;
+        }
+      }
+    }
+    if (request instanceof CancellableDependency && ((CancellableDependency) request).isCancelled()) {
+      return false;
+    }
+
+    // Retry if the request is considered idempotent
+    return handleAsIdempotent(request);
+  }
+
+  @Override
+  public boolean retryRequest(
+    final HttpResponse response,
+    final int execCount,
+    final HttpContext context) {
+    Args.notNull(response, "response");
+
+    return execCount <= this.maxRetries && retriableCodes.contains(response.getCode());
+  }
+
+  @Override
+  public TimeValue getRetryInterval(
```

Comment:
responses with 503 status code can include `Retry-After` header, technically it could hang on this too, not sure if spotify api ever returns 503 with `Retry-After` header. Maybe it's better to just simply return `this.defaultRetryInterval`?

Reply:
Hmm, so the client currently hangs because it waits for the amount of seconds specified in the  `Retry-After` header?

Reply:
And should the `503` handling be removed as for `429` as well then?

Reply:
> Hmm, so the client currently hangs because it waits for the amount of seconds specified in the `Retry-After` header?

Yes, using current method logic, if `Retry-After` header exists, it would return time to wait from header, otherwise return `this.defaultRetryInterval`. Then it would wait for amount which was returned.


> And should the `503` handling be removed as for `429` as well then?

No, it's safe to retry on 503.

Reply:
Ok, great! So are you happy with my latest changes? I'd merge the MR then :)